pkgs/servers/web-apps/plausible/update.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p jq nix-prefetch-git yarn yarn2nix-moretea.yarn2nix moreutils
+
+# NOTE: please check on new releases which steps aren't necessary anymore!
+# Currently the following things are done:
+#
+# * Add correct `name`/`version` field to `package.json`, otherwise `yarn2nix` fails to
+#   find required dependencies.
+# * Keep `tailwindcss` on version 2.0.1-compat (on `yarn` it will be upgraded due to the `^`).
+#   This is needed to make sure the entire build still works with `postcss-7` (needed
+#   by plausible).
+# * Adjust `file:`-dependencies a bit for the structure inside a Nix build.
+# * Update hashes for the tarball & the fixed-output drv with all `mix`-dependencies.
+# * Generate `yarn.lock` & `yarn.nix` in a temporary directory.
+
+set -x
+
+dir="$(realpath $(dirname "$0"))"
+latest="$(curl -q https://api.github.com/repos/plausible/analytics/releases/latest \
+  | jq -r '.tag_name')"
+nix_version="$(cut -c2- <<< "$latest")"
+
+if [ "$(nix-instantiate -A plausible.version --eval | xargs echo)" = "$nix_version" ];
+then
+  echo "Already using version $latest, skipping"
+  exit 0
+fi
+
+SRC="https://raw.githubusercontent.com/plausible/analytics/${latest}"
+
+package_json="$(curl -qf "$SRC/assets/package.json")"
+fixed_tailwind_version="$(jq '.dependencies.tailwindcss' -r <<< "$package_json" | sed -e 's,^^,,g')"
+
+echo "$package_json" \
+  | jq '. + {"name":"plausible","version":"'$latest'"}' \
+  | jq '.dependencies.tailwindcss = "'"$fixed_tailwind_version"'"' \
+  | sed -e 's,../deps/,../../tmp/deps/,g' \
+  > $dir/package.json
+
+tarball_meta="$(nix-prefetch-git https://github.com/plausible/analytics --rev "$latest" --quiet)"
```

Comment:
This seems somewhat error-prone: I wouldn't rely on `nix-prefetch-git` producing the same output as the tarball fetched by `fetchFromGitHub` — and it's very easy to miss if this does happen when updating, because of course it'll build successfully on the system where nix-prefetch-git ran! I'd prefer `nix-prefetch-url --unpack https://github.com/plausible/analytics/archive/"$latest".tar.gz --print-path` to get the hash and path.

Reply:
Using `nix-prefetch-github` now :)